packages/zk-transfer-web/src/index.tsx

```diff
@@ -3,12 +3,15 @@ import ReactDOM from "react-dom";
 import "./styles/app.css";
 import "./styles/tailwind.min.css";
 import App from "./App";
+import Offline from './components/Offline';
 import * as serviceWorker from "./serviceWorker";
 
 document.addEventListener('DOMContentLoaded', async () => {
     const { demo } : any = window;
-    
-    ReactDOM.render(<App demo={demo}/>, document.getElementById("root"));        
+
+    const Component = (demo) ? <App demo={demo} /> : <Offline />;
```

Comment:
Nice! It seems this deals with the problem where HTTP(S) requests from custom ports like 7001 sometimes get filtered out by firewalls.

However, in your case, without the VPN, the user is not technically offline. Otherwise they would never be able to load the page in the first place! We'll leave it for a future issue how to notify the user in that case, or to only use standard ports like 80 and 443 for both `ETH_URL` and `DB_URL`.